cfme/utils/template/rhevm.py

```diff
@@ -96,8 +95,10 @@ def templatize_vm(self):
         """
         self.mgmt.get_vm(self.temp_vm_name).mark_as_template(
             template_name=self.template_name,
-            cluster=self.provider_data.template_upload.cluster,
-            delete=False)  # leave vm in place in case it fails, for debug
+            cluster_name=self.provider_data.template_upload.cluster,
+            storage_domain_name=self.provider_data.template_upload.get('template_domain', None),
+            delete=False  # leave vm in place in case it fails, for debug
```

Comment:
so if template succeeds, the vm will be removed, right?

Reply:
VM cleanup is handled below on L115, if the VM still exists.